qcodes/instrument_drivers/rigol/DS4000.py

```diff
@@ -250,3 +252,11 @@ def __init__(self, name, address, timeout=20, **kwargs):
 
         channels.lock()
         self.add_submodule('channels', channels)
+
+    def _check_firmware_version(self):
+        #Require version 00.02.03
+
+        idn = self.get_idn()
+        ver = [int(x) for x in idn['firmware'].split('.')]
```

Comment:
is firmware version always a three-number string `#.#.#`?

you do not have to reinvent the wheel here, just use `from distutils.version import LooseVersion` ([example here](https://stackoverflow.com/a/11887885) or [here](https://stackoverflow.com/a/6972866) or [even in QCoDeS](https://github.com/QCoDeS/Qcodes/blob/3ff371052bb90ffdcfef192f9f1a505b6ba50c58/qcodes/instrument_drivers/rohde_schwarz/RTO1000.py#L315)) .

Another thing you could use is `from pkg_resources import parse_version` function ([examples here](https://setuptools.readthedocs.io/en/latest/setuptools.html#specifying-your-project-s-version) and [here](https://stackoverflow.com/a/21065570)) but it is not included in the standard library of python.

Reply:
Ah good, I was looking for something similar, but I didn't found it, thanks!